app/javascript/packs/followButtons.js

```diff
@@ -207,7 +207,10 @@ function handleFollowButtonClick({ target }) {
   ) {
     const userStatus = document.body.getAttribute('data-user-status');
     if (userStatus === 'logged-out') {
-      showLoginModal();
+      showLoginModal({
+        secondary_source: determineSecondarySource(target),
```

Comment:
In this case, we can determine whether the secondary source is an action (like tags or podcast follow) or a user follow. At this point we do not break down action further as per the requirements and it will require more effort.